lib/stellar/client.rb

```diff
@@ -196,6 +196,59 @@ def change_trust(
       horizon.transactions._post(tx: envelope_base64)
     end
 
+    Contract(C::KeywordArgs[
+      tx_envelope: Stellar::TransactionEnvelope,
+      options: Maybe[{ skip_memo_required_check: C::Bool }]
+    ] => Any)
+    def submit_transaction(tx_envelope:, options: { skip_memo_required_check: false })
```

Comment:
The goal is to insert this new check into existing workflows so somebody using the SDK today will have this check added without any change to their application. This PR looks like it adds a new function that is independent of existing functions so it's not clear to me how this goal is fulfilled. Is that correct, or do the existing functions call this somehow?

I think the existing functions that call `horizon.transactions._post` probably need to be changed to call this function too.

The Go SDK was simpler and only needed changing the submit transaction function because all calls to submit already happened through one function. See stellar/go#2396. Unfortunately that doesn't seem to happen in the Ruby SDK.

Reply:
You're correct, `submit_transaction` is a new function. We could adjust the methods that call `horizon.transactions._post`, but these functions should be deprecated because they use the deprecated methods on `Transaction` that `TransactionBuilder` replaced. I'll add the deprecation warning in a separate PR.

Reply:
I filed an issue to update the examples to use `submit_transaction` so hopefully new users of the SDK will use it.

Reply:
Even if they're deprecated, we only just deprecated them in the last week so I think we should assume most people using the SDK will be using them and propagate these changes to them.

Reply:
I agree, I adjusted them to call `submit_transaction`.